pkg/chaosdaemon/tasks/pod_handler.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tasks
+
+import (
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaoserr"
+)
+
+var ErrNotFoundSysPID = chaoserr.NotFound("SysPID")
+var ErrNotPodPID = errors.New("pid is not PodPID")
+var ErrPodProcessMapNotInit = errors.New("PodProcessMap not init")
+
+type PodID string
+
+func (p PodID) ToID() string {
+	return string(p)
+}
+
+type ChaosOnPOD interface {
+	Injectable
+	Recoverable
+}
+
+type PodProcessMap struct {
+	m      map[PodID]SysPID
+	rwLock sync.RWMutex
+}
+
+func NewPodProcessMap() PodProcessMap {
+	return PodProcessMap{
+		m:      make(map[PodID]SysPID),
+		rwLock: sync.RWMutex{},
+	}
+}
+
+func (p *PodProcessMap) Read(podPID PodID) (SysPID, error) {
+	p.rwLock.RLock()
+	defer p.rwLock.RUnlock()
+	sysPID, ok := p.m[podPID]
+	if !ok {
+		return SysPID(0), ErrNotFoundSysPID
```

Comment:
`return SysPID(0), errors.WithStack(ErrNotFoundSysPID)`

Reply:
Refer this https://github.com/chaos-mesh/rfcs/blob/main/text/2021-09-27-refine-error-handling.md

Reply:
ErrNotFoundSysPID already have a stack. It is created by errors.New()

Reply:
Fixed here , I find most of error in this package can be write with a better generic format after golang version uodate to 1.18, so I will finally update in this PR after the bump PR merged.
An example:
```golang 1.17
// Now
func NotFound(type_name string) error {
	return errors.Errorf("%s not found.", type_name)
}
```

```golang 1.18
// golang 1.18
func NotFound[T any]() error {
	var inner T
	return errors.New(fmt.Sprintf("%T not found", inner))
}
```

Reply:
All fixed.